src/get-started/create-service.md

```diff
@@ -371,72 +362,109 @@ executed:
 Let’s define the codes of the above errors:
 
 ```rust
-#[derive(Debug, Fail)]
-#[repr(u8)]
+/// Error codes emitted by `TxCreateWallet` and/or `TxTransfer`
+/// transactions during execution.
+#[derive(Debug, ExecutionFail)]
 pub enum Error {
-    #[fail(display = "Wallet already exists")]
+    /// Wallet already exists.
+    ///
+    /// Can be emitted by `TxCreateWallet`.
     WalletAlreadyExists = 0,
-
-    #[fail(display = "Sender does not exist")]
+    /// Sender doesn't exist.
+    ///
+    /// Can be emitted by `TxTransfer`.
     SenderNotFound = 1,
-
-    #[fail(display = "Receiver does not exist")]
+    /// Receiver doesn't exist.
+    ///
+    /// Can be emitted by `TxTransfer`.
     ReceiverNotFound = 2,
-
-    #[fail(display = "Insufficient currency amount")]
+    /// Insufficient currency amount.
+    ///
+    /// Can be emitted by `TxTransfer`.
     InsufficientCurrencyAmount = 3,
-
-    #[fail(display = "Sender same as receiver")]
+    /// Sender same as receiver.
+    ///
+    /// Can be emitted by `TxTransfer`.
     SenderSameAsReceiver = 4,
 }
-
-// Conversion between service-specific errors and the standard error type
-// that can be emitted by transactions.
-impl From<Error> for ExecutionError {
-    fn from(value: Error) -> ExecutionError {
-        let description = format!("{}", value);
-        ExecutionError::with_description(value as u8, description)
-    }
-}
 ```
 
+Deriving the `ExecutionFail` trait here will make our errors generic and
+compatible with other error kinds used within Exonum blockchain, this macro
+is similar to the `failure::Fail`.
+
 ### Transaction Execution
 
-Every transaction in Exonum has business logic of the blockchain attached,
-which is encapsulated in the `Transaction` trait.
-This trait has `execute` method which contains logic applied to the
-storage when a transaction is executed.
+Above we've defined the interface of our service, but currently this interface
+has no implementation. Thus, there is no actual business logic attached to
+them. To fix this situation, we should declare our service, and then implement
+the `CryptocurrencyInterface` trait for it.
+
+The service declaration is similar to the usual structure declaration:
```

Comment:
I'd rewrite as "Service is a `struct` with some trait implementations" since service declaration *is* a structure declaration.